src/geom/prep/BasicPreparedGeometry.cpp

```diff
@@ -40,12 +40,20 @@ BasicPreparedGeometry::setGeometry(const geom::Geometry* geom)
 bool
 BasicPreparedGeometry::envelopesIntersect(const geom::Geometry* g) const
 {
+    if (g->getDimension() == 0) {
+        return baseGeom->getEnvelopeInternal()->intersects(*(g->getCoordinate()));
```

Comment:
Does this handle a MultiPoint?

Reply:
I'll have to test what the fastest way is. Maybe just trying a `dynamic_cast` to `Point`.

Reply:
The other possibility is to provide methods on Geometry which check Envelope intersection and containment.  That way Point geometries can optimize by avoiding creating Envelopes just for that check.  Essentially this converts the logic for checking for a Point geometry type into a virtual call.

This seems a bit cumbersome, but I can't think of any other way to avoid materializing an Envelope object when it isn't strictly needed.

Reply:
> The other possibility is to provide methods on Geometry which check Envelope intersection and containment.

That's a good idea.

Reply:
I'm surprised this didn't trigger a test error.  Perhaps more test coverage is needed?

Reply:
If envelope testing is moved into the Geometry class as a virtual, then that may open the way to eliminating envelopes on Points completely.  It seems like this would be a large potential memory savings?

Not sure this can happen in JTS as it stands, but GEOS has more flexibility.

Reply:
Another option is to make `Envelope` into more of an interface, and have a `PointEnvelope` that is backed by a single coordinate.

Reply:
Wouldn't that still require an `PointEnvelope` object to be instantiated?  Unless I guess `Point` implemented `IEnvelope`...